airbyte-integrations/connectors/destination-mssql-v2/src/main/kotlin/io/airbyte/integrations/destination/mssql/v2/MSSQLWriter.kt

```diff
@@ -0,0 +1,39 @@
+package io.airbyte.integrations.destination.mssql.v2
+
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.write.DestinationWriter
+import io.airbyte.cdk.load.write.StreamLoader
+import io.airbyte.integrations.destination.mssql.v2.config.MSSQLConfiguration
+import io.airbyte.integrations.destination.mssql.v2.config.MSSQLDataSourceFactory
+import io.github.oshai.kotlinlogging.KotlinLogging
+import jakarta.inject.Singleton
+import javax.sql.DataSource
+
+@Singleton
+class MSSQLWriter(
+    private val config: MSSQLConfiguration,
+    private val dataSourceFactory: MSSQLDataSourceFactory
+) : DestinationWriter {
+    override fun createStreamLoader(stream: DestinationStream): StreamLoader {
+        val dataSource = dataSourceFactory.getDataSource(config)
```

Comment:
We may want to do this in the constructor/init/post construct method.  Otherwise, we are creating a data source per stream, which means d x s connections instead of just d connections if the data source is shared between streams (preferred to limit connections).